### ISSUE TITLE
async gets formatted away for functions with labeled arguments

### ISSUE THREAD

**Issue by cknitt** [rescript-lang/syntax#674]
```rescript
let f = async (a, b) => a + b

let f = async (~a, ~b) => a + b
```
gets reformatted to 
```rescript
let f = async (a, b) => a + b

let f = (~a, ~b) => a + b
```

**Comment by cknitt** [comment]
Also
```rescript
let f = async a => a
```
gets formatted to

```rescript
let f = async (a) => a
```

Is this intentional? Because for non-async functions we are *not* adding parentheses automatically.

**Comment by mattdamon108** [comment]
Fixed #677
